kubespawner/spawner.py

```diff
@@ -198,14 +199,17 @@ def __init__(self, *args, **kwargs):
                     max_workers=self.k8s_api_threadpool_workers
                 )
 
+            # Set global kubernetes client configurations
+            # before reflector.py code runs
+            self._set_k8s_client_configuration()
+            self.api = shared_client('CoreV1Api')
```

Comment:
Since this is changing global state it might be worth adding a comment in https://github.com/jupyterhub/kubespawner/blob/c215e24c3816e06b87409a50ffef2125465d1c44/kubespawner/clients.py#L24 ?